java-client/src/main/java/org/opensearch/client/json/jackson/JsonValueParser.java

```diff
@@ -49,7 +49,7 @@
  * object (e.g. START_OBJECT, VALUE_NUMBER, etc).
  */
 class JsonValueParser {
-    private final JsonProvider provider = JsonProvider.provider();
+    private static final JsonProvider provider = JsonProvider.provider();
```

Comment:
Let's postpone the provider initialization till it is really needed? 

```suggestion
    private static class DefaultJsonProvider {
        private static final JsonProvider INSTANCE = JsonProvider.provider();
    }    
```

And than just replace `provider` with `DefaultJsonProvider.INSTANCE` ?